src/wix/WixToolset.Core/Link/AddDefaultSymbolsCommand.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information.
+
+namespace WixToolset.Core.Link
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using WixToolset.Data;
+    using WixToolset.Data.Symbols;
+
+    internal class AddDefaultSymbolsCommand
+    {
+        public static readonly string WixStandardInstallFolder = "INSTALLFOLDER";
+        public static readonly string WixStandardInstallFolderParent = "ProgramFiles6432Folder";
+        public static readonly string WixStandardInstallFolderReference = "Directory:INSTALLFOLDER";
+
+        public AddDefaultSymbolsCommand(FindEntrySectionAndLoadSymbolsCommand find, List<IntermediateSection> sections)
+        {
+            this.Find = find;
+            this.Sections = sections;
+        }
+
+        public IntermediateSection EntrySection { get; }
+
+        public IEnumerable<IntermediateSection> Sections { get; }
+
+        public FindEntrySectionAndLoadSymbolsCommand Find { get; }
+
+        public void Execute()
+        {
+            if (this.Find.EntrySection.Type != SectionType.Package)
+            {
+                // Only packages...for now.
+                return;
+            }
+
+            var symbols = this.Sections.SelectMany(s => s.Symbols);
+            var directorySymbols = symbols.OfType<DirectorySymbol>();
+            var referenceSymbols = symbols.OfType<WixSimpleReferenceSymbol>();
+
+            if (referenceSymbols.Any(s => s.SymbolicName == WixStandardInstallFolderReference)
+                && !directorySymbols.Any(d => d.Id.Id == WixStandardInstallFolder))
```

Comment:
I don't think you need to loop through all the directory symbols. The `this.Find.SymbolsByName` is a dictionary of symbols by their `Id`. You should be able to check the dictionary for the existence of the `WixStandardInstallFolder` symbol.